docs/adr/0007-allow-implementing-bindings-seperately.md

```diff
@@ -0,0 +1,98 @@
+
+Testing Implementation: [PR 1206](https://github.com/mozilla/uniffi-rs/pull/1206)
+
+## Context and Problem Statement
+All the binding generators currently live in the [`uniffi_bindgen`](../../uniffi_bindgen/src/bindings) crate. This creates the following difficulties:
+
+- All the bindings live in the `uniffi` repository, so the `uniffi` team has to maintain them (or at the very least review changes to them).
+- Any change to a specific binding generator requires a new `uniffi_bindgen` release for it to be accessible by consumers. Even if it doesn't impact any of the other bindings.
+- Some bindings require complex build systems to test. Including those build systems in `uniffi` would require developers to install those build systems, and CI to do the same. For example, any type of `gecko-js` bindings would require the mozilla-central build system to build and test.
+- We currently run all the tests for the bindings in our CI and through `cargo test`. This means that if one binding target gets outdated and fails, or if a developer doesn't have the needed libraries installed for one of the targets, the tests would fail.
+- It's currently difficult to support third-party developers writing bindings for languages the core team does not wish to maintain.
+
+Before [PR 1201](https://github.com/mozilla/uniffi-rs/pull/1201), it was also impossible to write new bindings that did not live in the [`uniffi_bindgen`](../../uniffi_bindgen/src/bindings) crate.
+
+This ADR proposes allowing third-party crates to implement binding generators. 
+## Decision Drivers
+
+* Support Firefox Desktop JavaScript binding generation
+* Testability, it should be easy for developers to test the bindings they care about. Without having to navigate and install unfamiliar libraries and build systems.
+* Developer experience, it should be easier to write and maintain a new binding generator than it currently is.
+* Releases, cutting releases for changes in one binding generator shouldn't harm another.
+
+**NOTE**: Version compatibility is handled in a [separate ADR](https://github.com/mozilla/uniffi-rs/pull/1203)
+
+## Considered Options
+
+* **[Option 1] Do nothing**
+
+    This means keeping everything as-is, and deciding that all binding generators (at least for now) should live under the `uniffi_bindgen` crate.
+
+* **[Option 2] Create a public API for external crates to implement their own bindings.**
+
+    Developers would have traits exposed they can leverage to implement binding generators that do not live in `uniffi_bindgen`. `uniffi_bindgen` would still handle generic tasks related to binding generation. This includes:
+
+    1. Parsing the UDL, and passing the `ComponentInterface` to the specific binding generator.
+    1. Parsing the `uniffi.toml` and passing the language specific configuration down to the binding generator.
+    1. Calling a generic `BindingGenerator::write_bindings` function that is implemented by the specific binding generator.
```

Comment:
This seems like less of a requirement and more an implementation detail.  I think you can remove it, since you describe this at the bottom.